### PR TITLE
Auto open generated files in the IDE when generation completes support

### PR DESCRIPTION
**Pull request by ElizabethOkerio**
This PR adds support for automatically opening the generated files in the IDE when generation of the files completes.

### CODE REVIEW

File: src/Views/ConfigODataEndpoint.xaml [hunk 1]
```diff
@@ -24,5 +24,12 @@
                   Margin="20, 5, 5, 0" IsEditable="True"
                   ItemsSource="{Binding Path=UserSettings.MruEndpoints}"
                   Text="{Binding Path=Endpoint, TargetNullValue='Enter your endpoint...' }"/>
+        <CheckBox
+            x:Name="OpenGeneratedFilesOnIDE"
+            Margin="20,15,10,5"
+            HorizontalAlignment="Stretch"
+            Content="Open Generated Files In IDE When Generation Completes."
```
Comment (odero) [inline src/Views/ConfigODataEndpoint.xaml]:
```suggestion
            Content="Open generated files in the IDE when generation completes."
```

File: src/Models/UserSettings.cs [hunk 2]
```diff
@@ -19,6 +19,8 @@ internal class UserSettings
 
         [DataMember]
         public ObservableCollection<string> MruEndpoints { get; private set; }
+        [DataMember]
+        public bool OpenGeneratedFilesOnIDE { get; set; } = false;
```
Comment (odero) [inline src/Models/UserSettings.cs]:
```suggestion
        public bool OpenGeneratedFilesInIDE { get; set; } = false;
```

File: src/Views/ConfigODataEndpoint.xaml [hunk 3]
```diff
@@ -24,5 +24,12 @@
                   Margin="20, 5, 5, 0" IsEditable="True"
                   ItemsSource="{Binding Path=UserSettings.MruEndpoints}"
                   Text="{Binding Path=Endpoint, TargetNullValue='Enter your endpoint...' }"/>
+        <CheckBox
+            x:Name="OpenGeneratedFilesInIDE"
+            Margin="20,15,10,5"
+            HorizontalAlignment="Stretch"
+            Content="Open generated files in the IDE when generation completes."
+            FontWeight="Bold"
+            IsChecked="{Binding Path=UserSettings.OpenGeneratedFilesInIDE, Mode=TwoWay}" />
```
Comment (paulodero) [inline src/Views/ConfigODataEndpoint.xaml]: I feel this option should not be made available in the `Endpoint `tab but rather as an advanced settings

File: src/Views/ConfigODataEndpoint.xaml [hunk 4]
```diff
@@ -24,5 +24,12 @@
                   Margin="20, 5, 5, 0" IsEditable="True"
                   ItemsSource="{Binding Path=UserSettings.MruEndpoints}"
                   Text="{Binding Path=Endpoint, TargetNullValue='Enter your endpoint...' }"/>
+        <CheckBox
+            x:Name="OpenGeneratedFilesInIDE"
+            Margin="20,15,10,5"
+            HorizontalAlignment="Stretch"
+            Content="Open generated files in the IDE when generation completes."
+            FontWeight="Bold"
```
Comment (paulodero) [inline src/Views/ConfigODataEndpoint.xaml]: Any reason making this option bold (makes it look more important than other options which I do not think is the case)?

File: src/Views/ConfigODataEndpoint.xaml.cs [hunk 5]
```diff
@@ -6,14 +6,19 @@
 
 namespace Microsoft.OData.ConnectedService.Views
 {
+
+    public delegate void OpenGeneratedFilesInIDE(object sender, RoutedEventArgs e);
```
Comment (odero) [inline src/Views/ConfigODataEndpoint.xaml.cs]:
```suggestion
    public delegate void OpenGeneratedFilesInIDEHandler(object sender, RoutedEventArgs e);
```
c# naming conventions

**Comment by unchase** [comment]
Why this option allows only for OData v4 in `AdvancedSettings.xaml` (not for both v3 and v4 in `ConfigODataEndpoint.xaml`)?

**Comment by ElizabethOkerio** [comment]
@unchase we thought it wise to add this option as a setting in <code> AdvancedSettings.xaml </code> and not part of the endpoint. The <code>
ConfigODataEndPoint.xaml> will only have settings that configure the endpoint.

**Comment by unchase** [comment]
@ElizabethOkerio, in this case, you must enable the option for OData v 3 as well.

**Comment by ElizabethOkerio** [comment]
@unchase. yes thanks will do that.

**Pull request merged**